internal/tiltfile/tiltfile_test.go

```diff
@@ -1776,23 +1779,41 @@ docker_build('test/mycrd-env', 'env')
 	)
 }
 
-func TestK8SKind(t *testing.T) {
-	f := newFixture(t)
-	f.setupCRD()
-	f.dockerfile("env/Dockerfile")
-	f.dockerfile("builder/Dockerfile")
-	f.file("Tiltfile", `k8s_yaml('crd.yaml')
-k8s_kind('Environment', image_json_path='{.spec.runtime.image}')
+func testK8SKind(k8sKindArgs string, expectMatch bool) func(t *testing.T) {
+	return func(t *testing.T) {
+		f := newFixture(t)
+		f.setupCRD()
+		f.dockerfile("env/Dockerfile")
+		f.dockerfile("builder/Dockerfile")
+		f.file("Tiltfile", fmt.Sprintf(`k8s_yaml('crd.yaml')
+k8s_kind(%s)
 docker_build('test/mycrd-env', 'env')
-`)
+`, k8sKindArgs))
+
+		if expectMatch {
+			f.load("mycrd-env")
+			f.assertNextManifest("mycrd-env",
+				db(
+					image("docker.io/test/mycrd-env"),
+				),
+				k8sObject("mycrd", "Environment"),
+			)
+		} else {
+			w := unusedImageWarning("docker.io/test/mycrd-env", []string{"docker.io/library/golang"})
+			f.loadAssertWarnings(w)
+		}
+	}
+}
 
-	f.load("mycrd-env")
-	f.assertNextManifest("mycrd-env",
-		db(
-			image("docker.io/test/mycrd-env"),
-		),
-		k8sObject("mycrd", "Environment"),
-	)
+func TestK8SKind(t *testing.T) {
+	t.Run("match kind", testK8SKind("'Environment', image_json_path='{.spec.runtime.image}'", true))
```

Comment:
idiomatically, you'd usually do this with a table?
https://github.com/golang/go/wiki/TableDrivenTests

i don't think i've ever seen this style